otaclient/app/boot_control/_rpi_boot.py

```diff
@@ -161,6 +162,32 @@ def _init_boot_files(self):
             / f"{cfg.INITRD_IMG}{self.SEP_CHAR}{self.standby_slot}"
         )
 
+    def _update_firmware(self):
+        """Call external flash_kernel system script to install new firmware.
+
+        NOTE: this script also flashes kernel and initrd.img to /boot/firmware besides
+        dtb files, so we should also handle that.
+        """
+        try:
+            # call flash-kernel to install new dtb files, boot firmwares and kernel, initrd.img
+            # from current rootfs to system-boot partition
+            _cmd = "flash-kernel"
+            subprocess_call(_cmd, raise_exception=True)
+            os.sync()
```

Comment:
NOTE:
flash-kernel 
copies /boot/vmlinuz-5.15.0-1012-raspi to /boot/firmware/vmlinuz and
copies /boot/initrd.img-5.15.0-1012-raspi to /boot/firmware/initrd.img
and other number of stuffs.